### PR TITLE
PEP8 in reconst #881

### LINKED ISSUE #881: PEP8 in reconst
dipy/reconst/**init**.py:1:1: E265 block comment should start with '# '
dipy/reconst/**init**.py:12:1: W391 blank line at end of file
dipy/reconst/base.py:25:18: E225 missing whitespace around operator
dipy/reconst/base.py:27:34: E231 missing whitespace after ','
dipy/reconst/base.py:30:1: E302 expected 2 blank lines, found 1
dipy/reconst/cache.py:3:1: E302 expected 2 blank lines, found 1
dipy/reconst/csdeconv.py:94:80: E501 line too long (81 > 79 characters)
dipy/reconst/csdeconv.py:126:45: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dipy/reconst/csdeconv.py:142:80: E501 line too long (80 > 79 characters)
dipy/reconst/csdeconv.py:167:27: E221 multiple spaces before operator
dipy/reconst/csdeconv.py:183:5: E303 too many blank lines (2)
dipy/reconst/csdeconv.py:257:80: E501 line too long (81 > 79 characters)
dipy/reconst/csdeconv.py:278:45: E712 comparison to False should be 'if cond is False:' or 'if not cond:'
dipy/reconst/csdeconv.py:294:80: E501 line too long (80 > 79 characters)
dipy/reconst/csdeconv.py:399:1: E302 expected 2 blank lines, found 1
dipy/reconst/csdeconv.py:464:80: E501 line too long (80 > 79 characters)
dipy/reconst/csdeconv.py:572:13: E129 visually indented line with same indent as next logical line
dipy/reconst/csdeconv.py:590:80: E501 line too long (88 > 79 characters)
dipy/reconst/csdeconv.py:592:80: E501 line too long (92 > 79 characters)
dipy/reconst/csdeconv.py:628:80: E501 line too long (81 > 79 characters)
dipy/reconst/csdeconv.py:642:16: E203 whitespace before ':'
dipy/reconst/csdeconv.py:648:5: E265 block comment should start with '# '
dipy/reconst/csdeconv.py:658:13: E122 continuation line missing indentation or outdented
dipy/reconst/csdeconv.py:705:80: E501 line too long (80 > 79 characters)
dipy/reconst/csdeconv.py:818:80: E501 line too long (92 > 79 characters)
dipy/reconst/dsi.py:29:80: E501 line too long (116 > 79 characters)
dipy/reconst/dsi.py:203:80: E501 line too long (96 > 79 characters)
dipy/reconst/dsi.py:209:80: E501 line too long (86 > 79 characters)
dipy/reconst/dsi.py:243:80: E501 line too long (167 > 79 characters)
dipy/reconst/dsi.py:246:80: E501 line too long (92 > 79 characters)
dipy/reconst/dsi.py:251:80: E501 line too long (86 > 79 characters)
dipy/reconst/dsi.py:504:80: E501 line too long (174 > 79 characters)
dipy/reconst/dsi.py:505:80: E501 line too long (163 > 79 characters)
dipy/reconst/dsi.py:506:80: E501 line too long (166 > 79 characters)
dipy/reconst/dsi.py:649:80: E501 line too long (112 > 79 characters)
dipy/reconst/dti.py:1094:80: E501 line too long (94 > 79 characters)
dipy/reconst/dti.py:1786:5: E265 block comment should start with '# '
dipy/reconst/dti.py:1789:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:40:1: W293 blank line contains whitespace
dipy/reconst/gqi.py:44:1: E101 indentation contains mixed spaces and tabs
dipy/reconst/gqi.py:44:1: W191 indentation contains tabs
dipy/reconst/gqi.py:44:66: W291 trailing whitespace
dipy/reconst/gqi.py:45:1: E101 indentation contains mixed spaces and tabs
dipy/reconst/gqi.py:45:74: W291 trailing whitespace
dipy/reconst/gqi.py:46:38: W291 trailing whitespace
dipy/reconst/gqi.py:56:74: W291 trailing whitespace
dipy/reconst/gqi.py:78:32: E261 at least two spaces before inline comment
dipy/reconst/gqi.py:112:18: E225 missing whitespace around operator
dipy/reconst/gqi.py:113:17: E265 block comment should start with '# '
dipy/reconst/gqi.py:114:80: E501 line too long (152 > 79 characters)
dipy/reconst/gqi.py:116:80: E501 line too long (160 > 79 characters)
dipy/reconst/gqi.py:123:39: W291 trailing whitespace
dipy/reconst/gqi.py:142:71: W291 trailing whitespace
dipy/reconst/gqi.py:143:51: W291 trailing whitespace
dipy/reconst/gqi.py:168:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:173:80: E501 line too long (130 > 79 characters)
dipy/reconst/gqi.py:174:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:176:14: E231 missing whitespace after ','
dipy/reconst/gqi.py:176:17: E231 missing whitespace after ','
dipy/reconst/gqi.py:176:20: E231 missing whitespace after ','
dipy/reconst/gqi.py:184:20: E231 missing whitespace after ','
dipy/reconst/gqi.py:184:64: E231 missing whitespace after ','
dipy/reconst/gqi.py:184:80: E501 line too long (105 > 79 characters)
dipy/reconst/gqi.py:192:20: E231 missing whitespace after ','
dipy/reconst/gqi.py:192:64: E231 missing whitespace after ','
dipy/reconst/gqi.py:192:80: E501 line too long (105 > 79 characters)
dipy/reconst/gqi.py:204:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:206:80: E501 line too long (94 > 79 characters)
dipy/reconst/gqi.py:216:28: E231 missing whitespace after ','
dipy/reconst/gqi.py:220:20: E231 missing whitespace after ','
dipy/reconst/gqi.py:220:64: E231 missing whitespace after ','
dipy/reconst/gqi.py:220:80: E501 line too long (105 > 79 characters)
dipy/reconst/gqi.py:225:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:227:80: E501 line too long (91 > 79 characters)
dipy/reconst/gqi.py:242:5: E265 block comment should start with '# '
dipy/reconst/gqi.py:244:80: E501 line too long (91 > 79 characters)
dipy/reconst/gqi.py:251:38: E231 missing whitespace after ','
dipy/reconst/gqi.py:255:52: E502 the backslash is redundant between brackets
dipy/reconst/gqi.py:256:47: E127 continuation line over-indented for visual indent
dipy/reconst/gqi.py:257:64: E231 missing whitespace after ','
dipy/reconst/gqi.py:257:80: E501 line too long (96 > 79 characters)
dipy/reconst/gqi.py:264:31: E231 missing whitespace after ','
dipy/reconst/gqi.py:264:50: E231 missing whitespace after ','
dipy/reconst/interpolate.py:1:80: E501 line too long (87 > 79 characters)
dipy/reconst/interpolate.py:13:1: E302 expected 2 blank lines, found 1
dipy/reconst/interpolate.py:16:1: E302 expected 2 blank lines, found 1
dipy/reconst/interpolate.py:22:1: E302 expected 2 blank lines, found 1
dipy/reconst/interpolate.py:34:1: E302 expected 2 blank lines, found 1
dipy/reconst/mapmri.py:23:74: W291 trailing whitespace
dipy/reconst/mapmri.py:26:76: W291 trailing whitespace
dipy/reconst/mapmri.py:55:80: E501 line too long (81 > 79 characters)
dipy/reconst/mapmri.py:60:80: E501 line too long (81 > 79 characters)
dipy/reconst/mapmri.py:61:78: W291 trailing whitespace
dipy/reconst/mapmri.py:64:80: W291 trailing whitespace
dipy/reconst/mapmri.py:82:72: W291 trailing whitespace
dipy/reconst/mapmri.py:93:80: E501 line too long (83 > 79 characters)
dipy/reconst/mapmri.py:97:80: E501 line too long (83 > 79 characters)
dipy/reconst/mapmri.py:111:80: E501 line too long (129 > 79 characters)
dipy/reconst/mapmri.py:305:80: E501 line too long (99 > 79 characters)
dipy/reconst/mapmri.py:322:80: E501 line too long (87 > 79 characters)
dipy/reconst/mapmri.py:325:80: E501 line too long (91 > 79 characters)
dipy/reconst/mapmri.py:410:80: E501 line too long (81 > 79 characters)
dipy/reconst/mapmri.py:610:80: E501 line too long (82 > 79 characters)
dipy/reconst/odf.py:8:1: E302 expected 2 blank lines, found 1
dipy/reconst/odf.py:70:1: W391 blank line at end of file
dipy/reconst/peaks.py:7:1: E402 module level import not at top of file
dipy/reconst/sfm.py:127:80: E501 line too long (80 > 79 characters)
dipy/reconst/sfm.py:129:1: E302 expected 2 blank lines, found 1
dipy/reconst/sfm.py:178:80: E501 line too long (81 > 79 characters)
dipy/reconst/sfm.py:227:80: E501 line too long (81 > 79 characters)
dipy/reconst/sfm.py:281:80: E501 line too long (81 > 79 characters)
dipy/reconst/sfm.py:295:80: E501 line too long (130 > 79 characters)
dipy/reconst/sfm.py:309:80: E501 line too long (80 > 79 characters)
dipy/reconst/sfm.py:344:13: E125 continuation line with same indent as next logical line
dipy/reconst/sfm.py:344:13: E128 continuation line under-indented for visual indent
dipy/reconst/sfm.py:391:65: E225 missing whitespace around operator
dipy/reconst/sfm.py:391:69: E203 whitespace before ':'
dipy/reconst/sfm.py:398:52: E127 continuation line over-indented for visual indent
dipy/reconst/shm.py:1040:80: E501 line too long (151 > 79 characters)
dipy/reconst/shore.py:21:1: E302 expected 2 blank lines, found 1
dipy/reconst/shore.py:210:80: E501 line too long (99 > 79 characters)
dipy/reconst/shore.py:245:17: W503 line break before binary operator
dipy/reconst/shore.py:263:80: E501 line too long (93 > 79 characters)
dipy/reconst/shore.py:271:80: E501 line too long (96 > 79 characters)
dipy/reconst/shore.py:382:80: E501 line too long (115 > 79 characters)
dipy/reconst/shore.py:383:80: E501 line too long (129 > 79 characters)
dipy/reconst/shore.py:384:80: E501 line too long (85 > 79 characters)
dipy/reconst/shore.py:385:80: E501 line too long (96 > 79 characters)
dipy/reconst/shore.py:419:20: E222 multiple spaces after operator
dipy/reconst/shore.py:439:80: E501 line too long (98 > 79 characters)
dipy/reconst/shore.py:450:80: E501 line too long (167 > 79 characters)
dipy/reconst/shore.py:453:80: E501 line too long (92 > 79 characters)
dipy/reconst/shore.py:464:24: E221 multiple spaces before operator
dipy/reconst/shore.py:465:56: E221 multiple spaces before operator
dipy/reconst/shore.py:465:59: E222 multiple spaces after operator
dipy/reconst/shore.py:465:80: E501 line too long (103 > 79 characters)
dipy/reconst/shore.py:585:79: E202 whitespace before ')'
dipy/reconst/shore.py:585:80: E501 line too long (83 > 79 characters)
dipy/reconst/shore.py:595:80: E501 line too long (87 > 79 characters)
dipy/reconst/shore.py:627:80: E501 line too long (88 > 79 characters)
dipy/reconst/shore.py:636:80: E501 line too long (80 > 79 characters)
dipy/reconst/shore.py:637:80: E501 line too long (94 > 79 characters)
dipy/reconst/shore.py:753:80: E501 line too long (91 > 79 characters)
dipy/reconst/shore.py:790:80: E501 line too long (91 > 79 characters)
dipy/reconst/utils.py:3:1: E302 expected 2 blank lines, found 1

### PR DESCRIPTION
**Pull request by theaverageguy**
Fixes #881

### CODE REVIEW

File: dipy/reconst/csdeconv.py [hunk 1]
```diff
@@ -139,7 +140,8 @@ def __init__(self, gtab, response, reg_sphere=None, sh_order=8, lambda_=1,
         else:
             self.sphere = reg_sphere
 
-        r, theta, phi = cart2sphere(self.sphere.x, self.sphere.y, self.sphere.z)
+        r, theta, phi = cart2sphere(
+            self.sphere.x, self.sphere.y, self.sphere.z)
```
Comment (arokem) [inline dipy/reconst/csdeconv.py]: I prefer the breakdown you did on line 294 below, with each one of these on its own line.

File: dipy/reconst/gqi.py [hunk 2]
```diff
@@ -170,10 +173,11 @@ def npa(self, odf, width=5):
     psi0 = t0[1] ** 2
     psi1 = t1[1] ** 2
     psi2 = t2[1] ** 2
-    npa = np.sqrt((psi0 - psi1) ** 2 + (psi1 - psi2) ** 2 + (psi2 - psi0) ** 2) / np.sqrt(2 * (psi0 ** 2 + psi1 ** 2 + psi2 ** 2))
-    #print 'tom >>>> ',t0,t1,t2,npa
+    npa = np.sqrt((psi0 - psi1) ** 2 + (psi1 - psi2) ** 2 + (psi2 - psi0)
+                  ** 2) / np.sqrt(2 * (psi0 ** 2 + psi1 ** 2 + psi2 ** 2))
```
Comment (arokem) [inline dipy/reconst/gqi.py]: I prefer not to have a line break right before a binary operator (`**`)

**Comment by arokem** [comment]
Thanks! Just a couple of small comments.

For future reference, please use "rebase", instead of merging from the master branch: 

http://nipy.org/dipy/devel/gitwash/development_workflow.html#rebasing-on-trunk

**Comment by theaverageguy** [comment]
Please check @arokem :)

File: dipy/reconst/gqi.py [hunk 3]
```diff
@@ -109,11 +110,13 @@ def odf(self, sphere):
...
-                #print self.gqi_vector.shape
-                self.gqi_vector = np.real(H(np.dot(self.model.b_vector,                                         sphere.vertices.T) * self.model.Lambda))
+                H = squared_radial_component
+                # print self.gqi_vector.shape
+                self.gqi_vector = np.real(H(np.dot(self.model.b_vector,
+                                                   sphere.vertices.T) * self.model.Lambda))
             if self.model.method == 'standard':
-                self.gqi_vector = np.real(np.sinc(np.dot(self.model.b_vector,                                   sphere.vertices.T) * self.model.Lambda / np.pi))
+                self.gqi_vector = np.real(np.sinc(np.dot(self.model.b_vector,
+                                                         sphere.vertices.T) * self.model.Lambda / np.pi))
```
Comment (arokem) [inline dipy/reconst/gqi.py]: This line is still too long (>80 characters). Do you have a text editor that tells you about PEP8 issues? I use Atom with a PEP8 linter: https://atom.io/packages/linter-pep8

File: dipy/reconst/gqi.py [hunk 4]
```diff
@@ -181,7 +185,7 @@ def equatorial_zone_vertices(vertices, pole, width=5):
     finds the 'vertices' in the equatorial zone conjugate
     to 'pole' with width half 'width' degrees
     """
-    return [i for i,v in enumerate(vertices) if np.abs(np.dot(v,pole)) < np.abs(np.sin(np.pi*width/180))]
+    return [i for i, v in enumerate(vertices) if np.abs(np.dot(v, pole)) < np.abs(np.sin(np.pi * width / 180))]
```
Comment (arokem) [inline dipy/reconst/gqi.py]: Same here - line's too long

File: dipy/reconst/gqi.py [hunk 5]
```diff
@@ -189,21 +193,22 @@ def polar_zone_vertices(vertices, pole, width=5):
     finds the 'vertices' in the equatorial band around
     the 'pole' of radius 'width' degrees
     """
-    return [i for i,v in enumerate(vertices) if np.abs(np.dot(v,pole)) > np.abs(np.cos(np.pi*width/180))]
+    return [i for i, v in enumerate(vertices) if np.abs(np.dot(v, pole)) > np.abs(np.cos(np.pi * width / 180))]
```
Comment (arokem) [inline dipy/reconst/gqi.py]: And here

File: dipy/reconst/gqi.py [hunk 6]
```diff
@@ -213,18 +218,19 @@ def equatorial_maximum(vertices, odf, pole, width):
     return eqvertmax, eqvalmax
 
 
-def patch_vertices(vertices,pole, width):
+def patch_vertices(vertices, pole, width):
     """
     find 'vertices' within the cone of 'width' degrees around 'pole'
     """
-    return [i for i,v in enumerate(vertices) if np.abs(np.dot(v,pole)) > np.abs(np.cos(np.pi*width/180))]
+    return [i for i, v in enumerate(vertices) if np.abs(np.dot(v, pole)) > np.abs(np.cos(np.pi * width / 180))]
```
Comment (arokem) [inline dipy/reconst/gqi.py]: And here

**Comment by theaverageguy** [comment]
Why did the checks fail though @arokem?

**Comment by arokem** [comment]
Looks like a typo that I missed in my review: https://travis-ci.org/nipy/dipy/jobs/117482040#L2334

Thank providence for those tests :-)

File: dipy/reconst/gqi.py [hunk 7]
```diff
@@ -170,10 +173,11 @@ def npa(self, odf, width=5):
     psi0 = t0[1] ** 2
     psi1 = t1[1] ** 2
     psi2 = t2[1] ** 2
-    npa = np.sqrt((psi0 - psi1) ** 2 + (psi1 - psi2) ** 2 + (psi2 - psi0) ** 2) / np.sqrt(2 * (psi0 ** 2 + psi1 ** 2 + psi2 ** 2))
-    #print 'tom >>>> ',t0,t1,t2,npa
+    npa = np.sqrt((psi0 - psi1) ** 2 + (psi1 - psi2) ** 2 + (psi2 - psi0) ** 2) 
+                  / np.sqrt(2 * (psi0 ** 2 + psi1 ** 2 + psi2 ** 2))
```
Comment (arokem) [inline dipy/reconst/gqi.py]: These two lines need another set of parentheses. Also, the `/` is a binary operator, so please don't start the line with it.

**Comment by arokem** [comment]
You should run the tests locally on your machine, before pushing up to Github.

**Comment by arokem** [comment]
@theaverageguy : any progress here?

**Comment by arokem** [comment]
I assume this PR has been abandoned. Feel free to reopen if you still want to work on this.

**Pull request closed without merging**